pallets/gear/src/tests.rs

```diff
@@ -2860,6 +2860,71 @@ fn test_different_waits_fail() {
     });
 }
 
+// TODO:
+//
+// introduce new tests for this in #1485
+#[test]
+fn test_requeue_after_wait_for_timeout() {
+    use demo_waiter::{Command, WASM_BINARY};
+
+    init_logger();
+    new_test_ext().execute_with(|| {
+        assert_ok!(Gear::upload_program(
+            RuntimeOrigin::signed(USER_1),
+            WASM_BINARY.to_vec(),
+            DEFAULT_SALT.to_vec(),
+            EMPTY_PAYLOAD.to_vec(),
+            100_000_000u64,
+            0u128
+        ));
+
+        let program_id = get_last_program_id();
+
+        run_to_next_block(None);
+
+        let duration = 10;
+        let payload = Command::SendAndWaitFor(duration, USER_1.into()).encode();
+        assert_ok!(Gear::send_message(
+            RuntimeOrigin::signed(USER_1),
+            program_id,
+            payload,
+            2_500_000_000,
+            0,
+        ));
+
+        // Fast forward blocks.
+        run_to_next_block(None);
+        let message_id = filter_event_rev(|e| match e {
+            Event::MessageEnqueued { id, .. } => Some(id),
+            _ => None,
+        });
```

Comment:
Why?
`get_last_message_id` triggers on last message enqueue event + user message sent event. You do the same here, so if you call this method before `run_to_next_block` it will catch required user message

Message id to be waited can be found be the other events, that demonstrate wait